### PR TITLE
1691 localize accident count by car type

### PR DESCRIPTION
**Pull request by ziv17**
#1691 localize accident count by car type
closes 1691

### CODE REVIEW

File: anyway/infographics_utils.py [hunk 1]
```diff
@@ -914,16 +888,16 @@ def percentage_accidents_by_car_type(involved_by_vehicle_type_data):
...
+                driver_types[VehicleCategory.CAR.value] += count
+            elif vehicle_type in VehicleCategory.LARGE.get_codes():
+                driver_types[VehicleCategory.LARGE.value] += count
+            elif vehicle_type in VehicleCategory.MOTORCYCLE.get_codes():
+                driver_types[VehicleCategory.MOTORCYCLE.value] += count
+            elif vehicle_type in VehicleCategory.BICYCLE_AND_SMALL_MOTOR.get_codes():
+                driver_types[VehicleCategory.BICYCLE_AND_SMALL_MOTOR.value] += count
             else:
-                driver_types["אחר"] += count
+                driver_types[VehicleCategory.OTHER.value] += count
```
Comment (idosavion) [inline anyway/infographics_utils.py:900]: I know it's not your code, but what do you think about doing it as follows?
```
for category in [VehicleCategory.CAR, VehicleCategory.CAR, .., other]
    if vehicle_type in category.get_codes():
           driver_types[VehicleCategory.CAR.value] += count
           break
```

where the else category can be a all_possible_values - all_categories

Comment (idosavion) [reply anyway/infographics_utils.py:900]: but totally optional

File: anyway/vehicle_type.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            logging.exception(f"VehicleType.get_display_name: {self}: no display string defined")
+            return "no display name defined"
+
+    #
+    # def get_display_name(self) -> str:
+    #     try:
+    #         return self.english_vehicle_type_display_names[self.value]
+    #     except (KeyError, TypeError):
+    #         logging.exception(f"VehicleType.get_display_name: {self.value}: no display string defined")
+    #         return "no display name defined"
```
Comment (idosavion) [inline anyway/vehicle_type.py:80]: can we remove?

File: anyway/vehicle_type.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            VC.OTHER: _("other"),
+        }
+        try:
+            return english_vehicle_type_display_names[self]
+        except (KeyError, TypeError):
+            logging.exception(f"VehicleType.get_display_name: {self}: no display string defined")
+            return "no display name defined"
+
+
+VC = VehicleCategory
```
Comment (idosavion) [inline anyway/vehicle_type.py:192]: why?

File: anyway/infographics_dictionaries.py [hunk 4]
```diff
@@ -54,7 +54,7 @@ def __missing__(self, key):
 
 segment_dictionary = smart_dict()
 
-# bogus calls to gettext to get pybabel extract to recofnize the strings
+# bogus calls to gettext to get pybabel extract to recognize the strings
```
Comment (idosavion) [inline anyway/infographics_dictionaries.py:57]: nice catch!

**Review by idosavion — APPROVED** [review]
great PR! thanks!

File: anyway/backend_constants.py [hunk 5]
```diff
@@ -10,19 +10,18 @@ class BackEndConstants(object):
...
     # For percentage_accidents_by_car_type function
-    CAR_VEHICLE_TYPES = [1, 12]
-    LARGE_VEHICLE_TYPES = [25, 11, 4, 18, 6, 2, 3, 5, 13, 7, 24, 14]
-    MOTORCYCLE_VEHICLE_TYPES = [8, 10, 19, 9]
-    BICYCLE_AND_SMALL_MOTOR_VEHICLE_TYPES = [15, 21, 23]
-    OTHER_VEHICLE_TYPES = [14, 17, 22, 16]
+    # CAR_VEHICLE_TYPES = [1, 12]
+    # LARGE_VEHICLE_TYPES = [25, 11, 4, 18, 6, 2, 3, 5, 13, 7, 24, 14]
+    # MOTORCYCLE_VEHICLE_TYPES = [8, 10, 19, 9]
+    # BICYCLE_AND_SMALL_MOTOR_VEHICLE_TYPES = [15, 21, 23]
```
[the remark this answers is not in the document]
Comment (ziv17) [reply anyway/backend_constants.py:24]: You are right... removing.

File: anyway/infographics_utils.py (hunk 1 above)
Comment (ziv17) [reply anyway/infographics_utils.py:900]: The `OTHER` category is not the complement of the categories here. (There are additional categories, and other is the rest of the whole group.

File: anyway/vehicle_type.py (hunk 2 above)
Comment (ziv17) [reply anyway/vehicle_type.py:80]: correct, removing.

File: anyway/vehicle_type.py (hunk 3 above)
Comment (ziv17) [reply anyway/vehicle_type.py:192]: Just a shorthand, but not really justified. Removed.

**Review by idosavion — APPROVED** [review]

**Pull request merged**